### PR TITLE
Pull out and test revocableSeed logic

### LINKED ISSUE #21: NativeSeed tests and refactor out business logic
https://github1s.com/shapeshift/web/blob/HEAD/src/context/WalletProvider/NativeWallet/NativeSeed.tsx#L20-L69

Pull out all this biz logic into a hook and write a test

### PR DESCRIPTION
**Pull request by bootsboogie**
## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #21

### CODE REVIEW

File: src/context/WalletProvider/NativeWallet/NativeSeed/hooks/useRevocableSeed/useRevocableSeed.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { EncryptedWallet } from '@shapeshiftoss/hdwallet-native/dist/crypto'
```
Comment (cjthompson) [inline src/context/WalletProvider/NativeWallet/NativeSeed/hooks/useRevocableSeed/useRevocableSeed.ts:1]:
```suggestion
import { crypto } from '@shapeshiftoss/hdwallet-native'
const { EncryptedWallet } = crypto
```

File: src/context/WalletProvider/NativeWallet/NativeSeed/hooks/useRevocableSeed/useRevocableSeed.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      setIsGenerating(false)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [encryptedWallet?.encryptedWallet, revocableSeed])
+
+  useEffect(() => {
+    generate()
+  }, [generate])
+
+  return { revocableSeed, loading: generating }
```
Comment (cjthompson) [inline src/context/WalletProvider/NativeWallet/NativeSeed/hooks/useRevocableSeed/useRevocableSeed.ts:47]: I think we have an opportunity here to make it a bit easier.

We could do
```ts
return {
  getSeed: async () => {
    try {
      await generate()
      return revocableSeed.proxy.seed
    } catch (e) {
      // seed has been revoked
      return null // or maybe we generate another seed
    },
    revoke: revockableSeed.revoke
}
```

Comment (bootsboogie) [reply src/context/WalletProvider/NativeWallet/NativeSeed/hooks/useRevocableSeed/useRevocableSeed.ts:47]: So you are thinking that the code consuming the hook we would need to do this outside of the hook? 
```js
const { getSeed, revoke } = useRevocableSeed()

useEffect(() => {
  void (async () => {
    const s = await getSeed()
    setSeed(s)
  })
}, [])
```
It seems like at that point maybe this should not be a hook and should just be a helper function.

Or we could allow the hook to handle the generation on mount and then we just update it to return something a little more simple
```js
return { seed: revocableSeed.proxy.seed, revoke: revocableSeed.revoke }
```

Comment (cjthompson) [reply src/context/WalletProvider/NativeWallet/NativeSeed/hooks/useRevocableSeed/useRevocableSeed.ts:47]: The whole point is to NOT put the seed into state variables, so I wouldn't do that. Since it's just a simple function, you can just do `getSeed` where you need it. If making it async is a problem, then you could still do the `useEffect(() => generate()` and return a function that returns the value or catches the error and returns null.

Once a Proxy is revoked it'll always throw an error.

Comment (bootsboogie) [reply src/context/WalletProvider/NativeWallet/NativeSeed/hooks/useRevocableSeed/useRevocableSeed.ts]: Ah yea that makes a lot of sense. Thank you and updated with new changes.

**Review by cjthompson — COMMENTED** [review]
Can you add a test to see that mnemonic is null after calling revoke?

**Comment by bootsboogie** [comment]
> Can you add a test to see that mnemonic is null after calling revoke?

Good call. I added the test.

**Review by elmutt — APPROVED** [review]

**Review by cjthompson — APPROVED** [review]
